artiq/language/environment.py

```diff
@@ -217,10 +217,20 @@ def __init__(self, unprocessed_arguments):
     def get(self, key, processor, group, tooltip):
         if key in self.unprocessed_arguments:
             r = processor.process(self.unprocessed_arguments[key])
+            self.unprocessed_arguments.pop(key)
```

Comment:
Is there no side effect to modifying ``self.unprocessed_arguments`` here?

Reply:
`ProcessArgumentManager` is created in every run, so there will be new unprocessed_arguments every time.

Reply:
I think no because each argument only called this function once during build stage.

Reply:
Can you avoid this kind of mutation anyway? It's quite a footgun that ``ProcessArgumentManager`` modifies the dictionary it was given at creation when you call ``get()`` later. Especially since the creation doesn't even occur in this file.

For example, you could add a ``_processed_arguments`` private set and add to it instead.